proposals/0022-validating-resource-container-elements.md

```diff
@@ -58,22 +58,42 @@ field is saved, and the source location is used in the diagnostic output.
 
 * Examples:
 ```
-RWBuffer<int> // valid
-RWBuffer<float> // valid
-RWBuffer<float4> // valid
 struct x {
 	int i;
 };
-RWBuffer<x> // valid
+
+struct a {
+   int aa;
+   int ab;
+};
+
+struct b {
+   x bx;
+};
+
+struct c {
+  a ca;
+  float4 cb;

```

Comment:
Interesting.  I updated by comment because I put `float4` in there, and realized that `vector<float, 4>` probably isn't a valid type here. Is it?

Reply:
It shouldn't be valid because c is not a homogenous type, having both float4 and ints inside struct a.